### PR TITLE
test_provider_add_with_bad_credentials

### PR DESCRIPTION
**Pull request by lehinevych**
{{pytest: -k test_provider_add_with_bad_credentials --long-running -v}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit ccc3d681686b0d373ccaaf23281bcc078875d4bb:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/infrastructure/test_providers.py:152:101: E501 line too long (121 > 100 characters)
```
:trollface:
*CFME QE Bot*

File: cfme/tests/infrastructure/test_providers.py [hunk 1]
```diff
@@ -149,7 +149,7 @@ def test_provider_add_with_bad_credentials(provider):
         with error.expected('Cannot complete login due to an incorrect user name or password.'):
             provider.create(validate_credentials=True)
     elif isinstance(provider, RHEVMProvider):
-        with error.expected('401 Unauthorized'):
+        with error.expected('Credential validation was not successful: Login failed due to a bad username or password.'):
```
Comment (psav) [inline cfme/tests/infrastructure/test_providers.py]: Just a check that this is also present in 5.4, we may have to verpick it

Comment (lehinevych) [inline cfme/tests/infrastructure/test_providers.py]: Yes, you're right

**Comment by dajoRH** [comment]
Lint report for commit f39e7398874dea1c9903ad5374fa7a38540db282:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/infrastructure/test_providers.py [hunk 2]
```diff
@@ -149,7 +149,13 @@ def test_provider_add_with_bad_credentials(provider):
         with error.expected('Cannot complete login due to an incorrect user name or password.'):
             provider.create(validate_credentials=True)
     elif isinstance(provider, RHEVMProvider):
-        with error.expected('401 Unauthorized'):
+        with error.expected(
+            version.pick(
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_providers.py]: I'd rather recommend to pick the message before the with, this does not look so nice.

**Comment by dajoRH** [comment]
Lint report for commit 70327296857ef916f949374b227ee1c1f7f7feb0:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ac4a60d085b513ed2987367de17ea77287bbe229:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**